test/examples/boostrap.js

```diff
@@ -59,38 +52,18 @@ function waitServerReady(port) {
     });
 }
 
-function findFreeTcpPort(startPort) {
-    function _test(port, resolve) {
-        const client = net.createConnection({ port });
-        client.on('connect', () => {
-            client.end();
-            _test(port + 1, resolve);
-        });
-        client.on('error', () => {
-            resolve(port);
-        });
-    }
-    return new Promise((resolve) => {
-        _test(startPort, resolve);
-    });
-}
-
 before(async () => {
     let port = 8080;
 
     if (!process.env.USE_DEV_SERVER) {
-        port = await findFreeTcpPort(port);
+        global.itownsServer = startStaticFileServer();
+        port = global.itownsServer.address().port;
```

Comment:
IIRC, documentation says you should wait for the `listening` event to fire.

How about some more refactoring?
* let `startStaticFileServer()` assign `global.itownsServer`
* also let it listen for the `listening` event and assign `global.itownsPort` there (and also do the `console.log`); and return a promise that resolves when the event is dispatched (meaning the server accepts requests, replacing the need for `_waitServerReady` in this case)
* have `waitServerReady()` only be called when `USE_DEV_SERVER` is used, `await`ing `startStaticFileServer` otherwise

```js
if (!process.env.USE_DEV_SERVER) {
  await startStaticFileServer();
} else {
  global.itownsPort = process.env.USE_DEV_SERVER;
  await waitServerReady(global.itownsPort);
}
```

Alternatively, have `createStaticFileServer`'s returned promise resolve to the port it's listening on, and `waitServerReady` resolve to the passed in port; so you can assign `global.itownsPort` at a single location:
```js
let server;
if (!process.env.USE_DEV_SERVER) {
  server = startStaticFileServer();
} else {
  server = waitServerReady(process.env.USE_DEV_SERVER);
}
// wait for the server to be ready
global.itownsPort = await server;